react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import ReactDOMClient from 'react-dom/client';
+
+class NewsLetterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('state: ', this.state);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  render() {
+    return (
+      <form onSubmit = {this.handleSubmit}>
+        <label htmlFor="email"> Email </label>
+        <input type="text" id="email" onChange={this.handleChange} />
```

Comment:
Be sure to also pass a `value` prop to your input so that it is fully "controlled".

Reply:
Gotcha, value prop has been added to the input